hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java

```diff
@@ -1522,6 +1518,25 @@ private synchronized void updateJobStatus(String jobKey,
     snapDiffJobTable.put(jobKey, snapshotDiffJob);
   }
 
+  private synchronized void updateJobStatusToFailed(String jobKey,
+                                                    String reason) {
+    SnapshotDiffJob snapshotDiffJob = snapDiffJobTable.get(jobKey);
+    if (snapshotDiffJob.getStatus() != IN_PROGRESS) {
```

Comment:
could we cancel QUEUED job?

Reply:
No, that should not happen. If happens, we are invalid state.

Flow is Queue -> In_Progress/Rejected -> Done/Failed/Cancelled.
1. Once job is queue, we check if executor can take more job if yes, then change the state to  In_Progress otherwise Rejected.
2. In_Progress job is marked Done if report is generated successfully. In case of failure, job is marked Failed. Cancel is marked if someone execute the cancel request only if job is f In_Progress otherwise cancel request fails.

Reply:
I see, thanks Hemant!